### ISSUE TITLE
Odd Behavior Updating Display with TFT File - Driver v1.07

### ISSUE THREAD

**Issue by JeffHochberg** [on7lds/NextionDriver#8]
Hi! Amazing work in 1.07!!!

I was able to update the display (mostly). I renamed a TFT file with the name of the display model then copied it to the DataFilesPath (/etc on Pi-Star). I noticed that only the screens pertaining to a particular mode are updated. When there’s no activity, the display reverts back to the former screen loaded. But as soon as I see activity via D-Star/DMR/NXDN/POCSAG, I see the screens from the TFT file. 

Is this expected behavior? Or should the home screen (status) also be updated?

Again, this is when the Nextion Display is attached to ‘modem’ (Zumspot with RPi 3B). 

Thanks so much for incorporating the change!

**Comment by on7lds** [comment]
When a Nextion display is loaded with a TFT file, it works or it does not work. Since the TFT is a compiled file, it is loaded or it fails. It does not load just some screens.
When loading a new TFT, the display should go all white and display the progress of the loading (in %) until it is totally loaded or fails. Did this happen ?
If it loaded correctly, it seems to me that the TFT file is a version where someone has updated only some screens.
Where did you get the TFT ? Do you also have the HMI file ? When you open the HMI with the Nextion Editor, you should be able to check the screen layouts.

**Comment by JeffHochberg** [comment]
Hello, sorry for the delayed response. I created the TFT file by editing an HMI file I downloaded from Facebook. It was created by Rob Rheenan - he seems to be amongst the best of the screen creators on the Nextion Display FB group. I made small modifications to the HMI file to include my callsign, then ensured the 2.4" Enhanced display type was selected and ran the compiler. The compiler processed the HMI file with zero errors. Then I grabbed the TFT file from the directory in which Nextion Editor places them. I used WinSCP to transfer the file over to the home directory and renamed the file (by using 'mv'). Then used 'sudo' to copy the file from my home directory into /etc (the data files directory on Pi-Star). I simply rebooted the hotspot at that point. I gave the hotspot a few minutes to stabilize once rebooted, then went to the display and touched one of the virtual buttons on the screen. I did not see the screen turn white, nor did I see a progress indicator. The screen simply started displaying the new screens when displaying activity from a DMR station. When the transmission completed and the display went back to the home screen, the screen displayed the original screen format. I've attached a copy of my TFT file for you to test with.
[NX3224K024.zip](https://github.com/on7lds/NextionDriver/files/2701708/NX3224K024.zip)

**Comment by JeffHochberg** [comment]
I checked the Nextion Editor and it places the TFT files in C;\Users\%USERNAME%\AppData\Roaming\Nextion Editor\bianyi\

Am I grabbing the file from the wrong place?

**Comment by on7lds** [comment]
Rob's displays are very good, indeed :-)

The place of the tft seems correct (to be 100% sure : when, in the Nextion Editor, you choose **File** / **Open build folder** , the correct folder will be opened)
I can not test your TFT, since I have no K024 display. You could send your HMI.

As said, when updating, the display should be white and end with a success message, i.e. :
https://cdn.instructables.com/FKO/CNME/J6IEVLH4/FKOCNMEJ6IEVLH4.LARGE.jpg

Note: updating a display which is not connected directly to the Pi via NextionDriver is not supported at this moment.

The sceeen buttons only work with NextionDriver. Are you sure the NextionDriver is loaded and configured correctly ?

**Comment by JeffHochberg** [comment]
OK - then the issue is likely due to the fact the display is not connected directly to the Pi. The display is connected to a Zumspot MMDVM modem which of course is attached to the GPIO on the Pi. I guess it's possible that the power and I/O are not simply passed through the Zumspot to the GPIO.

But I am positive the NextionDriver is loaded and it is indeed version 1.07. I even built an entirely new install of Pi-Star on a new SD card. Then I followed your instructions to clone the NextionDriverInstaller to the SD card as well as allowed the script to modify the mmdvmhost file to ensure the required attribute/values were set correctly.

I spent several hours working on it to make sure I wasn't missing something. But at the end of the day, it sounds like the way the hardware is connected is the issue.

Thank you for getting back to me and thank you again for your work!!!
